src/Tools/RoboFile.php

```diff
@@ -192,4 +193,140 @@ private function endsWith($haystack, $needle) {
+    */
+   protected function getTrackedFiles($version = null) {
+      $output = [];
+      if ($version === null) {
+         $version = 'HEAD';
+      }
+      exec("git ls-tree -r '$version' --name-only", $output, $retCode);
+      if ($retCode != '0') {
+         throw new Exception("Unable to get tracked files");
+      }
+      return $output;
+   }
+
+   /**
+    * Update source code header in a source file
+    * @param string $filename
+    */
+   protected function replaceSourceHeader($filename) {
+      // get the content of the file to update
+      $source = file_get_contents($filename);
+
+      // define regex for the file type
+      $ext = pathinfo($filename, PATHINFO_EXTENSION);
+      switch ($ext) {
+         case 'php':
+            $source = $this->replacesourceHeaderForPHP($source);
+            break;
+
+         default:
+            // Unhandled file format
+            return;
+      }
+
+      if (file_put_contents($filename, $source) === false) {
+         throw new Exception('Failed to write ' . $filename);
+      }
+   }
+
+   /**
+    * Update the header in given source and returns the result
+    * @return string
+    */
+   protected function replacesourceHeaderForPHP($source) {
+      $prefix              = "\<\?php\\n/\*(\*)?\\n";
+      $replacementPrefix   = "<?php\n/**\n";
+      $suffix              = "\\n( )?\*/";
+      $replacementSuffix   = "\n */";
+
+      // format header template for the file type
+      $header = trim($this->getHeaderTemplate());
+      $formatedHeader = $replacementPrefix . $this->getFormatedHeaderTemplate('php', $header) . $replacementSuffix;
+
+      // update authors in formated template
+      $headerMatch = [];
+      $originalAuthors = [];
+      $authors = [];
+      $authorsRegex = "#^.*(\@author .*)$#Um";
+      preg_match('#^' . $prefix . '(.*)' . $suffix . '#Us', $source, $headerMatch);
+      if (isset($headerMatch[0])) {
+         $originalHeader = $headerMatch[0];
+         preg_match_all($authorsRegex, $originalHeader, $originalAuthors);
+         if (isset($originalAuthors[1])) {
+            $originalAuthors = $this->getFormatedHeaderTemplate('php', implode("\n", $originalAuthors[1]));
+            $formatedHeader = preg_replace($authorsRegex, $originalAuthors, $formatedHeader, 1);
+         }
+      }
+
+      // replace the header if it exists
+      $source = preg_replace('#^' . $prefix . '(.*)' . $suffix . '#Us', $formatedHeader, $source, 1);
+      if (empty($source)) {
+         throw new Exception("An error occurred while processing $filename");
+      }
+
+      return $source;
+   }
+
+   /**
+    * Read the header template from a file
+    * @throws Exception
+    * @return string
+    */
+   protected function getHeaderTemplate() {
+      if (empty($this->headerTemplate)) {
+         //$this->headerTemplate = file_get_contents(__DIR__ . '../../../../../tools/HEADER');
+         $this->headerTemplate = file_get_contents('tools/HEADER');
```

Comment:
Path will depends on location script is calle from. Probably not a good idea.

Also; tools provide a template HEADER file that may be used in fallback when a local HEADER is not present.